martin-tile-utils/src/lib.rs

```diff
@@ -33,6 +36,37 @@ impl Display for TileCoord {
     }
 }
 
+impl TileCoord {
+    /// Checks provided coordinates for validity
+    /// before constructing [`TileCoord`] instance.
+    #[must_use]
+    pub fn checked(z: u8, x: u32, y: u32) -> Option<TileCoord> {
+        Self::are_valid(z, x, y).then_some(Self { z, x, y })
+    }
+
+    /// Checks provided coordinates for validity
+    /// before constructing [`TileCoord`] instance with inverted `y` value.
+    #[must_use]
+    pub fn checked_inverted(z: u8, x: u32, y: u32) -> Option<TileCoord> {
+        Self::are_valid(z, x, y).then_some(Self {
+            z,
+            x,
+            y: (1_u32 << z) - 1 - y,
+        })
+    }
+
+    /// Checks if provided tile coordinates are valid.
+    #[must_use]
+    pub fn are_valid(z: u8, x: u32, y: u32) -> bool {
```

Comment:
Being valid is a bit vague.
Let's be explicit here instead what this asserts.

```suggestion
    /// Checks that zoom `z` is plausibily small and `x`/`y` is possible on said zoom level
    #[must_use]
    pub fn is_possible_on_zoom_level(z: u8, x: u32, y: u32) -> bool {
```